### ISSUE TITLE
[Suggestion] Permission Request method for iOS

### ISSUE THREAD

**Issue by pitAlex** [phonegap/phonegap-plugin-push#2085]
Hello,
In my current app, I show a custom dialog when user first runs the app to ask him if he wants to receive or not notifications. Its only when he accepts that I initiate the plugin. However this setup forces me to use this plugin : https://github.com/katzer/cordova-plugin-local-notifications
that has a "requestPermision" method which will tells me if user has allowed notifications or not when the system dialog for notifications shows up. Is there any plans to implement such functionality? It would definitely help me out not longer rely on the additional plugin as currently the register event doesn't tell you if user has allowed it or not.

**Comment by macdonst** [comment]
@pitAlex do you mean this method?

https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/API.md#pushnotificationhaspermissionsuccesshandler

**Comment by pitAlex** [comment]
Thats different. It will just tell you if permissions are granted or not. The method I am talking about actually fires the iOS system "notifications enable" dialog and waits for user's response before triggering the callback.
To make thing clear on my scenario. I have a custom overlay and if user "accepts" then the system dialog shows up because I am calling .requestPermission of that plugin.
In the phonegap-plugin-push, that dialog shows up if we do .init(), but should the user deny I did not see any callback telling me that.
I saw a similar thing referenced here : https://github.com/phonegap/phonegap-plugin-push/issues/2004
It was marked as duplicate, but its what I am experiencing. Either the registration event should not fire until user has selected an option or a new method like the "requestPermission" should be added.

**Comment by fredgalvao** [comment]
> Either the registration event should not fire until user has granted/denied permission or a new method like the "requestPermission" should be added.

Agreed, and I'd say that the former is a more sensible way to think about it. I'd even say it should have been like that from the start. `</my50cent>`

**Comment by gecsbernat** [comment]
Hi!
I have a solution for this.
https://github.com/phonegap/phonegap-plugin-push/pull/2843

It enables automatic permission request when checkpermission called. 
Simple way :)
